pkg/distributor/distributor.go

```diff
@@ -508,7 +511,13 @@ func (d *Distributor) createShard(streamshardCfg *shardstreams.Config, stream lo
 	}
 
 	shardLabel := strconv.Itoa(shardNumber)
-	lbls[len(lbls)-1] = labels.Label{Name: ingester.ShardLbName, Value: shardLabel}
+	for i := 0; i < len(lbls); i++ {
+		if lbls[i].Name == ingester.ShardLbName {
+			lbls[i].Value = shardLabel
+			break
+		}
+	}
+
 	return logproto.Stream{
 		Labels:  strings.Replace(streamPattern, ingester.ShardLbPlaceholder, shardLabel, 1),
```

Comment:
hmm maybe we don't need this anymore after the change of line 514?

Reply:
iirc, we did this to save allocations. labels -> string is pretty expensive. Instead, we generate a template once and do a string replacement.